docs/DeveloperGuide.md

```diff
@@ -141,12 +141,57 @@ Classes used by multiple components are in the `seedu.clinic.commons` package.
 
 This section describes some noteworthy details on how certain features are implemented.
 
-{To be updated with individual enhancement}
-
-### \[Proposed\] Data archiving
-
-_{Explain here how the data archiving feature will be implemented}_
-
+### Find feature
+
+#### What Find feature does
+The find feature allows users to find all relevant suppliers or warehouses by their names, and/or by their remarks
+and/or by the products sold/stored. Users are able to search for relevant suppliers or warehouses using either only one
+of these criterion or a combination of these criteria. Note that users are only able to search for either suppliers or
+warehouses at any one time and not both at the same time.
+
+#### How it is implemented
+Step 1. After the `find` command is called, the user input will be sent to **FindCommandParser** for parsing.
+
+Step 2. **FindCommandParser** will then check if the compulsory prefix `ct/COMMAND_TYPE` is present. If the user enters
+`ct/COMMAND_TYPE` prefix more than once, only the last prefix specified will be used to process user's input. If the
+prefix `ct/COMMAND_TYPE` is not present, a **ParseException** will be thrown. 
+
+Step 3. **FindCommandParser** will then proceed to check for the existence of at least one of the following prefixes
+ `n/NAME`, `r/REMARK` and `pd/PRODUCT`. If none is found, a **ParseException** will be thrown. Again, if the user
+ specifies the same prefix more than once, only the last prefix specified will be used to process the user's input.
+ 
+Step 4. Once the user has entered the correct format for the command, their input will then be parsed.
+
+Step 5. **FindCommandParser** will create a new **FindCommand** to be executed and the relevant suppliers or warehouses
+will be filtered out.
+
+Step 6. The model will then display the relevant suppliers or warehouses to the users via the method
+`model#getFilteredSupplierList()` or `model#getFilteredWarehouseList()`.
+
+#### Why it is implemented this way
+The `find` command is implemented this way to ensure consistency with the other commands in the application regarding
+the prefixes. This helps to minimise any potential confusion for the users by standardising the prefixes that `find`
+command takes in with the other relevant commands. In addition, a command type prefix, `ct/COMMAND_TYPE` is required in the
+implementation of `find` command to indicate whether the user wishes to search for suppliers or warehouses. Without this
+prefix, the application will not be able to know if the user wishes to search for suppliers or warehouse.
```

Comment:
I feel the prefix part looks a bit similar to the edit command's reasoning (and maybe can be used in other commands as well), will it be too repetitive?

Reply:
Yes I used the same reasoning as edit! But then again, this was why we decided to go with prefix for the find command as well!